src/DGtal/shapes/MeshTextureHelpers.h

```
@@ -0,0 +1,501 @@
+    using VertexNormalMesh = typename TriangulatedSurf::template IndexedPropertyMap<RealPoint>;
+    
+    ///Texture image
+    using TextureImage = ImageContainerBySTLVector<Z2i::Domain, Color>;
+    
+    
+    
+    /// Convert Per Corner normal vectors (from the OBJ reader) to
+    /// on normal vector per vertex (by avering them).
+    /// @param mesh input mesh
+    /// @param normalMesh normalMesh property map (corner based vectors)
+    /// @param normalMap normal vector buffer
+    static
+    VertexNormalMesh computeNormalPerVertex(const TriangulatedSurf  & mesh,
+                                            const NormalMesh &normalMesh,
+                                            const NormalMap &normalMap)
+    {
+      auto vertexNorm = mesh.template makeVertexMap<RealPoint>({0.,0.,0.});
+      for(auto f=0; f < mesh.nbFaces(); ++f)
+      {
+        auto verts = mesh.verticesAroundFace(f);
+        auto norms = normalMesh[f];
+        vertexNorm[ verts[0] ] += normalMap[ norms[0] ];
+        vertexNorm[ verts[1] ] += normalMap[ norms[1] ];
+        vertexNorm[ verts[2] ] += normalMap[ norms[2] ];
+      }
+      
+      for(auto v=0; v < mesh.nbVertices(); ++v)
+        if (mesh.isVertexBoundary(v))
+          vertexNorm[v] /= (mesh.degree(v) -1);
+        else
+          vertexNorm[v] /= (mesh.degree(v));
+
+      return vertexNorm;
+    }
+    
+    
+    static
+    std::tuple<TriangulatedSurf,  UVMesh, NormalMesh, UVMap, NormalMap>
+    loadOBJWithTextureCoord(const std::string &filename)
+    {
+      TriangulatedSurf mesh;
+
+      UVMap textureMap;
+      NormalMap normalMap;
+      UVMesh textMesh;
+      NormalMesh normalMesh;
+
+      std::vector<UVTriangle> originalVertIndices;
+      std::vector<UVTriangle> originalUVIndices;
+      std::vector<UVTriangle> originalNormalIndices;
+
+      std::ifstream in(filename, std::ios::in);
+      if (!in)
+      {
+        std::cerr << "Cannot open " << filename << std::endl;
+        exit(1);
```

That look like a bad idea. Why not to throw an exception?